src/main/java/com/exactpro/th2/act/ActHandler.java

```diff
@@ -297,30 +298,66 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
         }
     }
 
+    private void createAndStoreNoResponseEvent(String actName, Map<String, CheckMetadata> expectedMessages, String fieldValue,
+                                               Instant start,
+                                               EventID parentEventId, List<MessageID> messageIDList) throws JsonProcessingException {
+
+        Event errorEvent = Event.from(start)
+                .endTimestamp()
+                .name(format("Internal %s error", actName))
+                .type("No response found by target keys.")
+                .status(FAILED)
+                .bodyData(createNoResponseBody(expectedMessages, fieldValue));
+        for(MessageID msgID : messageIDList){
+            errorEvent.messageID(msgID);
+        }
+        storeEvent(errorEvent.toProtoEvent(parentEventId.getId()));
+    }
+
+    private IBodyData createNoResponseBody(Map<String, CheckMetadata> expectedMessages, String fieldValue) {
+        TreeTableBuilder treeTableBuilder = new TreeTableBuilder();
+        CollectionBuilder passedOn = new CollectionBuilder();
+        CollectionBuilder failedOn = new CollectionBuilder();
+        for (Map.Entry<String, CheckMetadata> entry : expectedMessages.entrySet()) {
+            if (entry.getValue().eventStatus == PASSED) {
+                passedOn.row(entry.getKey(), new CollectionBuilder().row(entry.getValue().fieldName, new RowBuilder().column(new EventUtils.MessageTableColumn(fieldValue)).build()).build());
+            }
+            if (entry.getValue().eventStatus == FAILED) {
```

Comment:
Without using else we always double check the eventStatus. This affects efficiency in terms of time complexity.